### PR TITLE
Qt6 port

### PR DESCRIPTION
**Pull request by tsujan**

### CODE REVIEW

**Comment by tsujan** [comment]
@stefonarch 
Please check all kinds of screenshots as well as `Edit in...` and saving under X11.

File: cmake/LXQtTranslateTs.cmake [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #       qmFiles The generated compiled translations (.qm) files
 #
 #     Input:
-#       USE_QT5 Optional flag to choose between Qt4 and Qt5. Defaults to Qt5
+#       USE_QT6 Optional flag to choose between Qt4 and Qt6. Defaults to Qt6
```
Comment (stefonarch) [inline cmake/LXQtTranslateTs.cmake:40]: Qt4→Qt5 here.

File: cmake/LXQtTranslateTs.cmake [hunk 2]
```diff
@@ -60,16 +60,16 @@ cmake_minimum_required(VERSION 2.8.3 FATAL_ERROR)
 
 
 function(lxqt_translate_ts qmFiles)
-    set(oneValueArgs USE_QT5 UPDATE_TRANSLATIONS TEMPLATE TRANSLATION_DIR INSTALL_DIR)
```
Comment (stefonarch) [inline cmake/LXQtTranslateTs.cmake:63]: Shouldn't this check too if Qt5 is set for compilation?

File: cmake/LXQtTranslateTs.cmake [hunk 3]
```diff
@@ -83,21 +83,21 @@ function(lxqt_translate_ts qmFiles)
...
                 ${TR_SOURCES}
                 ${tsFiles}
                 OPTIONS -locations absolute
             )
         else()
-            qt5_add_translation(QM ${tsFiles})
+            qt6_add_translation(QM ${tsFiles})
         endif()
     else()
         # Qt4
```
Comment (stefonarch) [inline cmake/LXQtTranslateTs.cmake:103]: Shouldn't this part be about qt5?

**Review by stefonarch — COMMENTED** [review]
Testing is all ok, didn't find any regressions.

**Comment by Chiitoo** [comment]
Similar to somewhere else, I hit a few header issues in `core.cpp` and in `extedit.h`:

```
src/core/core.cpp:33:10: fatal error: qt5xdg/XdgDesktopFile: No such file or directory
   33 | #include <qt5xdg/XdgDesktopFile>
      |          ^~~~~~~~~~~~~~~~~~~~~~~
```

Also, my mouse pointer does not seem to get included when the option is enabled, but I'm not sure if that worked before?

Thank you!

**Comment by stefonarch** [comment]
>Also, my mouse pointer does not seem to get included when the option is enabled, but I'm not sure if that worked before?

This did work before, tested under debian VM with 1.4 now. Retesting...

**Comment by tsujan** [comment]
@stefonarch 
Please check again. I removed "USE_QT..." and fixed the header of qt6xdg.

**Comment by Chiitoo** [comment]
Forgot to mention that was on KWin X11, and yeah, I see it now too, in another clean-user session.

This other session might be a bit... slightly messed up right now but I'll try to see what is going on with the cursor later.

**Comment by tsujan** [comment]
@Chiitoo 
Please test it again (I added a commit).

> Similar to somewhere else,...

Yes. Supporting two versions of Qt — even in the minimal way we do it (only for styling and file dialog) — can be confusing. It should be fixed now.

**Comment by Chiitoo** [comment]
> @Chiitoo Please test it again (I added a commit).
> Yes. Supporting two versions of Qt — even in the minimal way we do it (only for styling and file dialog) — can be confusing. It should be fixed now.

Indeed!

All good now here too.

**Comment by tsujan** [comment]
@Chiitoo 
Thanks! Your tests have saved us a lot of headache twice :)

**Review by stefonarch — APPROVED** [review]

**Comment by tsujan** [comment]
Will merge it tomorrow, after re-reading the changes.

**Pull request merged**